### PR TITLE
feat(route): add 中国智库网-观点与实践

### PR DESCRIPTION
**Pull request by AEliu**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/chinathinktanks/2
/chinathinktanks/3
/chinathinktanks/4
/chinathinktanks/5
/chinathinktanks/6
/chinathinktanks/7
/chinathinktanks/9
/chinathinktanks/10
/chinathinktanks/126
/chinathinktanks/127
/chinathinktanks/128
/chinathinktanks/129
/chinathinktanks/130
/chinathinktanks/131
/chinathinktanks/132
/chinathinktanks/133
/chinathinktanks/134
/chinathinktanks/135
/chinathinktanks/136
/chinathinktanks/137
/chinathinktanks/138
/chinathinktanks/139
/chinathinktanks/140
/chinathinktanks/141
/chinathinktanks/142
/chinathinktanks/143
/chinathinktanks/144
/chinathinktanks/145
/chinathinktanks/146
/chinathinktanks/147
/chinathinktanks/148
/chinathinktanks/149
/chinathinktanks/150
/chinathinktanks/151
/chinathinktanks/152
/chinathinktanks/153
/chinathinktanks/154
/chinathinktanks/155
/chinathinktanks/156
/chinathinktanks/157
/chinathinktanks/158
/chinathinktanks/181
```

## 新RSS检查列表 / New RSS Script Checklist
  
- [x] New Route
- [ ] Documentation
  - [x] CN
  - [ ] EN
- [x] 全文获取 fulltext
  - [x] Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] 日期和时间 date and time
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added 
- [ ] `Puppeteer`

## 说明 / Note
第一次 pr 献给你，现学现卖

### CODE REVIEW

**Review by TonyRL — CHANGES_REQUESTED** [review]
Please use v2 script standard https://github.com/DIYgod/RSSHub/discussions/8236

File: lib/v2/chinathinktanks/maintainer.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+    '/chinathinktanks/:id': ['Aeliu'],
```
Comment (TonyRL) [inline lib/v2/chinathinktanks/maintainer.js:2]:
```suggestion
    '/:id': ['AEliu'],
```

File: lib/v2/chinathinktanks/viewpoint.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    const out = await Promise.all(
+        list.map(async (info) => {
+            const title = info.title;
+            const pubdate = info.pubdate;
+            const itemUrl = info.link;
+
+            const cache = await ctx.cache.get(itemUrl);
+            if (cache) {
+                return Promise.resolve(JSON.parse(cache));
+            }
```
Comment (TonyRL) [inline lib/v2/chinathinktanks/viewpoint.js:33]: Please use `ctx.cache.tryGet` instead

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/v2/chinathinktanks/viewpoint.js (hunk 2 above)
Comment (TonyRL) [reply lib/v2/chinathinktanks/viewpoint.js:33]: You can take reference from the previous PR #8778
https://github.com/DIYgod/RSSHub/blob/63ba1db4f63c327a32a8c530cee0549617d435ef/lib/v2/cool18/index.js#L34-L54

**Comment by TonyRL** [comment]
Please click the "Resolve conversation" button if you have fixed the issue.

**Comment by AEliu** [comment]
> Please click the "Resolve conversation" button if you have fixed the issue.

OK, thank u!

**Comment by TonyRL** [comment]
Nothing to merge.
Please cherry-pick back 71de1017b04dc9765306a1362e331b13167b3ce5

File: lib/v2/chinathinktanks/viewpoint.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    const list = $('#relates > div.pub_right > div.pub_content > ul > li')
+        .map(function () {
+            const info = {
+                title: $(this).find('a').attr('title'),
+                link: `https:${$(this).find('a').attr('href')}`,
+                pubdate: $(this).find('span').text().slice(1, -1).trim(),
+            };
+            return info;
+        })
```
Comment (TonyRL) [inline lib/v2/chinathinktanks/viewpoint.js:21]:
```suggestion
        .map((info) => {
            info.title = $(this).find('a').attr('title');
            info.link = `https:${$(this).find('a').attr('href')}`;
            info.pubdate = $(this).find('span').text().slice(1, -1).trim();
            return info;
        })
```

**Review by TonyRL — CHANGES_REQUESTED** [review]

**Comment by TonyRL** [comment]
Since the target site is under maintenance, I'll revisit this once it is up again.

**Review by TonyRL — CHANGES_REQUESTED** [review]
The site is back online but it seems the layout has been changed quite a lot

**Pull request merged**